🧹 remove the remaining wordpress pieces

danyx23 opened the pull request:
Fixes #3565

mlbrgl commented on package.json:
```diff
@@ -35,8 +33,6 @@
         "startTscServer": "tsc -b -verbose -watch",
         "startViteServer": "vite dev",
-        "startWordpressPlugin": "cd ./wordpress/web/app/plugins/owid && yarn && yarn start",
-        "startSiteFront": "./devTools/vite/startVite.sh",
```
this is still used by make up and make up.full. I'll revert it now to be able to continue testing, but feel free to amend if you had something else in mind.

danyx23 replied:
Thanks, that was a mistake :)

mlbrgl commented on docker-compose.full.yml:
```diff
@@ -79,42 +59,8 @@ services:
             - 8080:80
         volumes:
             - ./devTools/docker/vhosts.conf:/etc/nginx/conf.d/default.conf
-            - ./wordpress:/app
         depends_on:
             - fpm
```
`web` references the `fpm` container and `vhosts.conf`, which both don't exist anymore. `web` is not needed at this point though, so we should just remove it.

mlbrgl commented on docs/setup-options-overview.md:
```diff
@@ -12,10 +12,6 @@ At a minimum, all of the options will transpile and serve the Grapher code, whic
 
 This needs a working MySQL database and gives you an interactive admin UI for visually editing our charts as well as all our ~4000 charts and their data.
 
-**Full WordPress setup** 🚀
```
if we're keeping `make up.full`, I think we should be coherent and explain here what this setup is for. However, t this point, the only difference I could notice between `make up` and `make up.full` was related to the cloudflare functions dev server. It used to be that the overhead of the `full` variant justified the split and associated maintenance, but this is much less true now, and we should just merge them and have a single local dev command.

mlbrgl reviewed (changes requested):
Almost there! 👏

I've fixed a couple of regressions, and a few things still need a bit of attention, but this is overall looking pretty nice!

I still need to go over images, and will confirm when done.

On the discussion front, we should consider whether keeping `make up.full` still makes sense, given the small delta with `make up` (see comment).

mlbrgl commented on adminSiteServer/mockSiteRouter.tsx:
```diff
@@ -379,16 +378,6 @@ mockSiteRouter.get("/headerMenu.json", async (req, res) => {
     res.send(await renderMenuJson())
 })
 
-mockSiteRouter.use(
-    // Not all /app/uploads paths are going through formatting
-    // and being rewritten as /uploads. E.g. blog index images paths
-    // on front page.
-    ["/uploads", "/app/uploads"],
-    express.static(path.join(WORDPRESS_DIR, "web/app/uploads"), {
-        fallthrough: false,
-    })
-)
-
```
images on the remaining wordpress pages will not render now that they are not available locally, so we should probably redirect to R2 (or document it so we know it is a wontfix)

danyx23 replied:
Good catch - I added a new route that redirects to assets.ourworldindata.org for wp images

mlbrgl reviewed (commented):
my understanding of what's left of the [wordpress images baking flow](https://github.com/owid/ops/blob/57b50b942abcb6959bf15b606bf8fae7410720b5/templates/owid-cloudflare-prod/README.md#wordpress-images-baking-flow):

<img width="486" alt="Screenshot 2024-05-02 at 21 40 30" src="https://github.com/owid/owid-grapher/assets/13406362/24fe9a3b-e240-4523-8ef2-90d826e42ae3">

The master copy of wordpress images is now R2, and updates to Wordpress uploads (/uploads) [should be made directly on R2](https://www.notion.so/owid/Editing-old-wordpress-posts-e-g-about-pages-6ea77093df0344b1a6c1bf5cfe47c51b?pvs=4#76ab5a35a5b9429b8dad38778c97797f).

I've finished my review, let me know if you want to discuss certain points.

danyx23 commented on Makefile:
```diff
@@ -137,8 +134,6 @@ migrate:
 	rm -rf itsJustJavascript && yarn && yarn buildLerna && yarn buildTsc && yarn runDbMigrations
 
-refresh.full: refresh refresh.wp sync-images
-
 refresh:
```
Nice idea - I added that

danyx23 commented on adminSiteClient/PostsIndexPage.tsx:
```diff
@@ -222,14 +222,6 @@ class PostRow extends React.Component<PostRowProps> {
                 <td>
                     <Timeago time={post.updatedAtInWordpress} />
```
Hilarious, this has been broken for a long time :)

danyx23 commented on baker/startDeployQueueServer.ts:
```diff
@@ -10,6 +10,7 @@ import * as db from "../db/db.js"
 // by registering listeners on SIGINT.
 import "../db/cleanup.js"
 
+// TODO: I think we can delete this script but I'm not entirely sure.
```
Right - thanks for clarifying! I added a comment to communicate why we still have this.

danyx23 commented on ops/buildkite/deploy-content:
```diff
@@ -55,16 +54,6 @@ update_owid_content_repo() {
     )
 }
 
-sync_wordpress_uploads() {
```
Removed it

danyx23 commented:
@mlbrgl thanks a lot for your thorough review, much appreciated! I have:
- added a `refresh.full` command
- removed the docker-compose.full.yaml file
- kept the "make up.full" command even if that only differs in the CF worker setup - but if we merge this into "make.up" then when you run it you are asked to authenticate to CF in a browser and that will confuse non-staff people running our setup
- added a redirect in mocksite router for old images so that images are fetched by the client from assets.ourworldindata.org
- removed some code you mentioned was redundant
- fixed the updated_at_in_wordpress bug :)
- and rebased on top of master

It would be great if you could have one last look at the last commit, then I think we could ship it!

mlbrgl reviewed (approved):
that all makes sense, thanks for update!

I've done a couple more cleanup tasks, and rebaked the whole site on staging. Everything looks good.
